consensus/spos/bls/sovereignSubRoundEnd.go

```diff
@@ -0,0 +1,118 @@
+package bls
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/sovereign"
+	"github.com/multiversx/mx-chain-go/consensus/spos"
+	"github.com/multiversx/mx-chain-go/errors"
+	"github.com/multiversx/mx-chain-go/process/block"
+)
+
+type sovereignSubRoundEnd struct {
+	*subroundEndRoundV2
+	outGoingOperationsPool block.OutGoingOperationsPool
+	bridgeOpHandler        BridgeOperationsHandler
+}
+
+// NewSovereignSubRoundEndRound creates a new sovereign end subround
+func NewSovereignSubRoundEndRound(
+	subRoundEnd *subroundEndRoundV2,
+	outGoingOperationsPool block.OutGoingOperationsPool,
+	bridgeOpHandler BridgeOperationsHandler,
+) (*sovereignSubRoundEnd, error) {
+	if check.IfNil(subRoundEnd) {
+		return nil, spos.ErrNilSubround
+	}
+	if check.IfNil(outGoingOperationsPool) {
+		return nil, errors.ErrNilOutGoingOperationsPool
+	}
+	if check.IfNil(bridgeOpHandler) {
+		return nil, errors.ErrNilBridgeOpHandler
+	}
+
+	sr := &sovereignSubRoundEnd{
+		subroundEndRoundV2:     subRoundEnd,
+		outGoingOperationsPool: outGoingOperationsPool,
+		bridgeOpHandler:        bridgeOpHandler,
+	}
+
+	sr.Job = sr.doSovereignEndRoundJob
+
+	return sr, nil
+}
+
+func (sr *sovereignSubRoundEnd) doSovereignEndRoundJob(ctx context.Context) bool {
+	success := sr.subroundEndRoundV2.doEndRoundJob(ctx)
+	if !success {
+		return false
+	}
+
+	sovHeader, castOk := sr.Header.(data.SovereignChainHeaderHandler)
+	if !castOk {
+		log.Error("sovereignSubRoundEnd.doSovereignEndRoundJob", "error", errors.ErrWrongTypeAssertion)
+		return false
+	}
+
+	outGoingMBHeader := sovHeader.GetOutGoingMiniBlockHeaderHandler()
+	if check.IfNil(outGoingMBHeader) {
+		return true
+	}
+
+	currBridgeData, err := sr.updateBridgeDataWithSignatures(outGoingMBHeader)
+	if err != nil {
+		return false
```

Comment:
You may need to log something here in case you get an error. or is it not needed?

Reply:
Right, very good point, added log error here